R/pkg/R/DataFrame.R

```diff
@@ -1880,4 +1880,46 @@ setMethod("as.data.frame",
               stop(paste("Unused argument(s): ", paste(list(...), collapse=", ")))
             }
             collect(x)
+          }
+)
+
+#' Returns the column types of a DataFrame.
+#' 
+#' @name coltypes
+#' @title Get column types of a DataFrame
+#' @param x (DataFrame)
+#' @return value (character) A character vector with the column types of the given DataFrame
+#' @rdname coltypes
+setMethod("coltypes",
+          signature(x = "DataFrame"),
+          function(x) {
+            # TODO: This may be moved as a global parameter
+            # These are the supported data types and how they map to
+            # R's data types
+            DATA_TYPES <- c("string"="character",
+                            "long"="integer",
+                            "tinyint"="integer",
+                            "short"="integer",
+                            "integer"="integer",
+                            "byte"="integer",
+                            "double"="numeric",
+                            "float"="numeric",
+                            "decimal"="numeric",
+                            "boolean"="logical"
+            )
```

Comment:
You only handle primitive types here, but no complex types, like Array, Struct and Map.

It would be better you can refactor the type mapping related code here and that in SerDe.